models/things.js

```diff
@@ -28,21 +30,38 @@ var Things = {
    /**
     * Get all Things stored in the database.
     *
-    * @return Promise which resolves with a list of Thing objects.
+    * @return {Promise} which resolves with a Map of Thing objects.
     */
    getThings: function() {
      return new Promise((function(resolve, reject) {
        Database.getThings().then((function(things) {
          // Update the map of Things
-         this.things = {};
+         this.things = new Map();
```

Comment:
@dhylands  Yes I wasn't sure whether NodeJS supported Map because it's a relatively recent feature of JavaScript and doesn't have universal support in browsers. They have both advantages and disadvantages though.